src/features/groups/slice.ts

```diff
@@ -42,3 +55,11 @@ export default groupsSlice.reducer;
 export const groupsListSelector = (store: RootState) => store.groups.list;
 export const selectGroupById = (groupId: string) => (store: RootState) => 
   store.groups.list.find(g => g.id === groupId);
+
+export const selectedGroupSelector = (store: RootState) => store.groups.selectedGroup ? ({
+  ...store.groups.selectedGroup,
+  transactions: store.groups.selectedGroup.transactions.map(t => ({
+    ...t,
+    paymentDateUtc: new Date(t.paymentDateUtc),
+  }))
+}) : undefined;
```

Comment:
Is any reason to use 'undefined', but not null?

Reply:
Just to make it a bit consistent with the state where we also use undefined instead of null
```
interface ISliceState {
  list: ITransactionGroupListItem[],
  selectedGroup?: ITransactionGroup,
}
```